add room struct with Message Adapter

jz709u opened the pull request.

jz709u committed:
add room struct with Message Adapter

furkanmustafa commented on CoThings/ViewController.swift:
```diff
@@ -332,3 +325,46 @@ class ViewController: UIViewController, CLLocationManagerDelegate, UITableViewDe
+
+
+
+class MessageAdapter {
```
This class sounds like it's better to be in a separate file. At least not in ViewController.

jz709u replied:
yeah I can do that

furkanmustafa replied:
Thanks a lot!

furkanmustafa commented on CoThings/ViewController.swift:
```diff
@@ -136,18 +136,11 @@ class ViewController: UIViewController, CLLocationManagerDelegate, UITableViewDe
-                        self.lobbyTable.reloadData()
-                    }
+                guard let response = MessageAdapter(message: data).decodeRoomResponse() else {
+                    return
```
Didn't check the whole code, but.. Do we have any way of error handling instead of ignoring? (discussion)

nesimtunc replied:
We don't have error handling mechanism yet, but better to have at the beginning, I'll also ask kindly to designer friends to show error cases in the app.

furkanmustafa replied:
Until we have a general flow of handling errors, we can just silence it here, and create a new issue for handling errors.

In that case it's better to leave a TODO comment here, and/or an issue number if possible. This would also allow merging this work without getting blocked on this.

furkanmustafa replied:
Going to be handled in this issue => https://github.com/rainlab-inc/cothings-ios/issues/2
resolving.

furkanmustafa commented on CoThings/ViewController.swift:
```diff
@@ -332,3 +325,46 @@ class ViewController: UIViewController, CLLocationManagerDelegate, UITableViewDe
+            let roomResponse = try decoder.decode(RoomsResponse.self, from: jsonData)
+            return roomResponse
+        } catch {
+            return nil
```
Feels like it's better to let it throw, and handle the error on the caller side.

Reason: the input provided here is obviously wrong/faulty. In such cases, throwing an exception is better IMO.

jz709u replied:
if we are throwing an error I think we probably be throwing a different error instead of the one thrown by decoding error.  like conversion error of some sort

furkanmustafa replied:
I agree, that makes more sense.

furkanmustafa replied:
I think last update also covers this, resolving!

jz709u committed:
Add Adapter Class around the SwiftPheonix library to make the apis
cleaner to deal with.

cleaned up view controller

added error handling for swift pheonix

jz709u commented:
wrapped the swift pheonix into an adapter class, reducing direct dependencies on that library.  please check

jz709u committed:
error codes fix

furkanmustafa commented:
Thanks @jz709u !

I don't understand swift much, so all my feedbacks were those three, and all three handled :)

@nesimtunc should review and merge.

nesimtunc commented:
looks ✨ I just wonder why _Swift Developers_ don't use plural names for folders. Ex: ~~Models~~ Model :)
